### DISCUSSION TITLE
Do we have any control over array format options?

### DISCUSSION THREAD

**Discussion by viveleroi** [prettier/prettier#14355] [Q&A]
Because format linters like eslint and stylelint are dropping format rules and recommending prettier, we're planning to move over. However, we've been very specific and have had the same format rules for a decade so I'm trying to configure prettier as closely as possible.

I've run through the config docs but I see absolutely no way for us to control arrays. We prefer to define arrays like this, with brackets on the same line - it's shorter and requires one less indentation:

```js
const options = [{
  value: 'kermit',
  text: 'Kermit'
}, {
  value: 'fozzy',
  text: 'Fozzy'
}, {
  value: 'animal',
  text: 'Animal'
}]
```

But prettier forces the array formatting to this:

```js
const options = [
  {
    value: 'kermit',
    text: 'Kermit'
  },
  {
    value: 'fozzy',
    text: 'Fozzy'
  },
  {
    value: 'animal',
    text: 'Animal'
  }
]
```

Worst case scenario, we'll adapt - but is there truly no way to configure this? Seems odd to me we don't get a choice.

**Comment by karfous** [comment]
Hello @viveleroi how did you manage to get the objects expanded to multiple lines? Curly braces alone on the row? I am not succesfull doing this.

I would also appreciate more control over arrays. For example like this. It was possible with "Prettier now" extension which is deprecated.
![image](https://user-images.githubusercontent.com/44132167/220205276-e50d4280-a5e5-4089-8abe-39d5ab2a7e11.png)

**Comment by awoimbee** [comment]
I think this would be useful for single value arrays, eg take a look at this kubernetes config snippet:
```diff
-spec: {
-  rules: [{
-    http: {
-      paths: [{
-        path: "/",
-        pathType: "Prefix",
-        backend: {
-          service: {
-            name: svc.metadata.apply(m => m.name),
-            port: { number: 80 }
-          }
-        }
-      }]
-    }
-  }]
-}
+spec: {
+  rules: [
+    {
+      http: {
+        paths: [
+          {
+            path: "/",
+            pathType: "Prefix",
+            backend: {
+              service: {
+                name: svc.metadata.apply((m) => m.name),
+                port: { number: 80 },
+              },
+            },
+          },
+        ],
+      },
+    },
+  ],
+},
``` 

Prettier adds so much whitespace to my pulumi code that it grows from ~3600 lines to ~4600 :(
